processor/processor.go

```diff
@@ -916,10 +1163,59 @@ func (proc *HandleT) mainLoop() {
 	}
 }
 
-func (proc *HandleT) crashRecover() {
+func (proc *HandleT) readErrJobsLoop() {
+	logger.Info("Processor errors backup loop started")
 
 	for {
-		execList := proc.gatewayDB.GetExecuting([]string{gateway.CustomVal}, dbReadBatchSize, nil)
+		time.Sleep(errReadLoopSleep)
+		proc.statErrDBR.Start()
+
+		toQuery := errDBReadBatchSize
+		retryList := proc.errorDB.GetToRetry(nil, toQuery, nil)
+		toQuery -= len(retryList)
+		unprocessedList := proc.errorDB.GetUnprocessed(nil, toQuery, nil)
+
+		proc.statErrDBR.End()
+
+		combinedList := append(retryList, unprocessedList...)
+
+		if len(combinedList) == 0 {
+			logger.Debug("[Processor: readErrJobsLoop]: DB Read Complete. No proc_err Jobs to process")
+			continue
+		}
+
+		hasFileUploader := proc.errFileUploader != nil
+
+		jobState := jobsdb.Executing.State
+		// abort jobs if file uploader not configured to store them to object storage
+		if !hasFileUploader {
+			jobState = jobsdb.Aborted.State
+		}
+
+		var statusList []*jobsdb.JobStatusT
+		for _, job := range combinedList {
+			status := jobsdb.JobStatusT{
+				JobID:         job.JobID,
+				AttemptNum:    job.LastJobStatus.AttemptNum + 1,
+				JobState:      jobState,
+				ExecTime:      time.Now(),
+				RetryTime:     time.Now(),
+				ErrorCode:     "",
+				ErrorResponse: []byte(`{}`),
+			}
+			statusList = append(statusList, &status)
+		}
+
+		proc.errorDB.UpdateJobStatus(statusList, nil, nil)
+		if hasFileUploader {
+			proc.errProcessQ <- combinedList
```

Comment:
How about just dumping to errorDB and let the errored jobs be backed up from jobsdb in a separate error_processor instead of our regular processor.